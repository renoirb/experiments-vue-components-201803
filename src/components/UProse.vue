<template>
  <div
    class="u-prose--component"
  >
    <component
      :is="currentType"
    >
<slot/>
    </component>
  </div>
</template>

<script>
import VueMarkdown from 'vue-markdown'

const ProseHtml = {
  render (h) {
    return h(
      'div',
      {
        class: {
          'u-prose-html': true,
        },
      },
      this.$slots.default
    )
  },
}

// @vue/component
export default {
  name: 'u-prose',
  props: {
    markdown: {
      type: Boolean,
      default: true,
    },
  },
  computed: {
    currentType () {
      return this.markdown ? 'markdown' : 'markup'
    },
  },
  components: {
    markdown: VueMarkdown,
    markup: ProseHtml,
  },
}
</script>

<style>
.u-prose--component {
  padding: 5px;
}
.u-prose--component ul,
.u-prose--component ol {
  margin-bottom: 15px;
}
.u-prose--component li {
  margin-left: 15px;
}
</style>
