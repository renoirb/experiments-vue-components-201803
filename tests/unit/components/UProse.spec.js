import {
  mount,
} from '@vue/test-utils'

import UProse from '@/components/UProse.vue'

describe(`${UProse.name}`, () => {
  it('Parses markdown', () => {
    const defaultSlotContent = `# Hello **World**\n`
    const wrapper = mount(UProse, {
      slots: {
        default: defaultSlotContent,
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
