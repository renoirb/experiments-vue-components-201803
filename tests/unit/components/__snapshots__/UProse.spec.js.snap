// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`u-prose Parses markdown 1`] = `
<div
  class="u-prose--component"
>
  <div>
    <h1>
      Hello 
      <strong>
        World
      </strong>
    </h1>
    

  </div>
</div>
`;
